picard/config_upgrade.py

```diff
@@ -342,6 +343,49 @@ def upgrade_to_v2_6_0_beta_3(config):
     _s.remove("use_system_theme")
 
 
+def upgrade_to_v2_7_0_dev_1(config):
+    """Replace manually set persistent splitter settings with automated system.
+    """
+    def _check_and_add(new_key, key_map):
+        _p = config.persist
+        splitter_dict = {}
+        for (oldkey, newkey) in key_map:
+            if _p.__contains__(oldkey):
+                if _p[oldkey] is not None:
+                    splitter_dict[newkey] = bytearray(_p[oldkey])
+                _p.remove(oldkey)
+        Option("persist", new_key, {})
+        _p[new_key] = splitter_dict
+
+    # MainWindow splitters
+    _check_and_add(
+        new_key="splitters_MainWindow",
+        key_map=[
+            ('bottom_splitter_state', '.QSplitter.MainWindow'),
+            ('splitter_state', 'MainPanel.QSplitter.MainWindow'),
+        ]
+    )
+
+    # ScriptEditorDialog splitters
+    _check_and_add(
+        new_key="splitters_ScriptEditorDialog",
+        key_map=[
+            ('script_editor_splitter_samples', '.splitter_between_editor_and_examples.ScriptEditor.RenamingOptionsPage.pages_stack.splitter.Dialog.MainWindow'),
+            ('script_editor_splitter_samples_before_after', '.splitter_between_before_and_after.groupBox.splitter_between_editor_and_examples.ScriptEditor.RenamingOptionsPage.pages_stack.splitter.Dialog.MainWindow'),
+            ('script_editor_splitter_documentation', '.splitter_between_editor_and_documentation.frame.frame_4.splitter_between_editor_and_examples.ScriptEditor.RenamingOptionsPage.pages_stack.splitter.Dialog.MainWindow'),
+        ]
```

Comment:
I wonder if we really need such "lineage", I see potential issues with that:
- when UI is modified, lineage may too, so saved config is lost, a new one will be create, but old one will stay in config file
- if we want to avoid that, we need to write upgrade functions after each UI change that may concern a splitter, that's not great
- isn't Objectname usable alone for that? If we enforce a unique and mandaroty name for each splitter, we don't need the whole lineage

Is this making sense?

Reply:
> Is this making sense?

Yes.  All good points.  I'll make the appropriate changes.

Reply:
It now uses objectName() value if it is available, otherwise it will use the lineage.  I also added names to the two main screen splitters.